lib/services/local/assistant.go

```diff
@@ -0,0 +1,213 @@
+/*
+ *
+ * Copyright 2023 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package local
+
+import (
+	"context"
+	"encoding/json"
+	"sort"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/lib/backend"
+)
+
+// Conversation is a conversation entry in the backend.
+type Conversation struct {
+	Title          string    `json:"title,omitempty"`
+	ConversationID string    `json:"conversation_id"`
+	CreatedTime    time.Time `json:"created_time"`
+}
+
+// CreateAssistantConversation creates a new conversation entry in the backend.
+func (s *IdentityService) CreateAssistantConversation(ctx context.Context,
```

Comment:
Why are these attached to `IdentityService`?

```
// IdentityService is responsible for managing web users and currently
// user accounts as well
```

This kind of stuff is hard to unwind, so getting the data model right seems important.

Reply:
To speed up the initial development, I decided to piggyback most of the implementation on `IdentityService`. I moved the implementation now.